### ISSUE TITLE
Fonts get blocked by CORS policy

### ISSUE THREAD

**Issue by Marchyello** [jsreport/jsreport-dotnet#27]
**Background:**
Generating pdf from .cshtml on linux using jsreport.AspNetCore middleware.

**jsreport package versions:**
AspNetCore - 2.0.0
Local - 2.0.0
Binary.Linux - 2.2.2

**Problem:**
Fonts are missing in generated pdf. The question is - how do I avoid that happening? Debug logs reveal that they are being blocked by CORS policy:
```
+0      Starting rendering request 2 (user: null)
+1      Rendering anonymous template { recipe: chrome-pdf, engine: none }
+1      Data item not defined for this template.
+1      Resources not defined for this template.
+3      Base url injected: http://127.0.0.1:5000
+3      Rendering engine none
+5      Compiled template not found in the cache, compiling
+6      Executing recipe chrome-pdf
+281    Access to font at 'http://127.0.0.1:5000/fonts/RobotoCondensed.ttf' from origin 'null' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
+282    Access to font at 'http://127.0.0.1:5000/fonts/Roboto-Bold.ttf' from origin 'null' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
+321    Chrome will wait for printing trigger
+328    Running chrome with params {"$id":"3","printBackground":true,"landscape":true,"format":"A4","marginTop":"0px","marginRight":"0px","marginBottom":"0px","marginLeft":"0px","waitForJS":true,"margin":{"top":"0px","right":"0px","bottom":"0px","left":"0px"}}
+398    Skipping storing report.
```
A bit weird thing here is that the .cshtml also contains image assets, but they are not getting blocked.

**Render call:**
```
[MiddlewareFilter(typeof(JsReportPipeline))]
...
{
    ...
    HttpContext.JsReportFeature()
        .Recipe(Recipe.ChromePdf)
        .Configure(c => c.Template.Chrome = ExportUtils.GetChrome())
        .OnAfterRender((r) =>
        {
            Response.Headers["Content-Disposition"] = $"attachment; filename=\"filename.pdf\"";
            Response.Headers.Add("access-control-expose-headers", "Content-Disposition");
        });

    return View("pdf", pdfData);
}
```
**Configuration:**
```
services.AddJsReport(new LocalReporting()
    .KillRunningJsReportProcesses()
    .UseBinary(jsBinary)
    .Configure(cfg =>
    {
        cfg.TempDirectory = Path.Combine(_env.ContentRootPath, "jsreport_tmp");
        cfg.AllowedLocalFilesAccess().FileSystemStore().BaseUrlAsWorkingDirectory();
        return cfg;
    })
    .AsUtility()
    .Create());
```

I stumbled upon some discussion that suggested that CORS options should be configured for Express extension. However, if that is the case `Configuration.ExtensionsConfiguration.ExpressConfiguration` contains only InputRequestLimit property, so I don't see how I should go about it.

**Comment by pofider** [comment]
I believe this url targets your asp.net web server right?
`http://127.0.0.1:5000/fonts/RobotoCondensed.ttf`
This means you need to configure CORS on your asp.net web server, not in jsreport.

**Comment by Marchyello** [comment]
Yes, it does target the server.
However, the web server CORS policy is configured as follows:
```
options.AddPolicy("CorsPolicy", new CorsPolicyBuilder()
    .AllowAnyHeader()
    .AllowAnyMethod()
    .AllowAnyOrigin()
    .AllowCredentials()
    .Build());
```
Shouldn't this cover the problem? Also as I mentioned the images are getting through (they're on the same path). In that case shouldn't they be blocked as well?

**Comment by pofider** [comment]
I believe the problem is in your cors configuration.

I tried this and it works for me
https://docs.microsoft.com/en-us/aspnet/core/security/cors?view=aspnetcore-2.2
```csharp
services.AddCors();            
....
app.UseCors(builder => builder.WithOrigins("*"));
```

**Comment by Marchyello** [comment]
I get the same exact error both when setting allowed origins to wildcard as you suggested: `WithOrigins("*")` (which I believe has the same effect as `AllowAnyOrigin()` that I was already using) as well as when setting them to local server explicitly: `WithOrigins("http://127.0.0.1:5000")`. 

I'm reverting to previous solution for now (inlining resouces) since this is a low priority issue, but nevertheless it would be nice to get to the bottom of this.
